vendor/diff-parsers-upstream-std.patch

```diff
@@ -0,0 +1,821 @@
+   | None, None -> assert false
+ 
+ let mkexp_opt_constraint ~loc e = function
+   | None -> e
+   | Some constraint_ -> mkexp_constraint ~loc e constraint_
+ 
+ let mkpat_opt_constraint ~loc p = function
+   | None -> p
+-  | Some typ -> ghpat ~loc (Ppat_constraint(p, typ))
++  | Some typ -> mkpat ~loc (Ppat_constraint(p, typ))
+ 
+ let syntax_error () =
+   raise Syntaxerr.Escape_error
+ 
+ let unclosed opening_name opening_loc closing_name closing_loc =
+@@ -387,16 +387,16 @@
+   loc_map Longident.last id
+ 
+ let loc_lident (id : string Location.loc) : Longident.t Location.loc =
+   loc_map (fun x -> Lident x) id
+ 
+-let exp_of_longident ~loc lid =
+-  let lid = make_ghost (loc_map (fun id -> Lident (Longident.last id)) lid) in
+-  ghexp ~loc (Pexp_ident lid)
++let exp_of_longident lid =
++  let lid = loc_map (fun id -> Lident (Longident.last id)) lid in
++  Exp.mk ~loc:lid.loc (Pexp_ident lid)
+ 
+-let exp_of_label ~loc lbl =
+-  mkexp ~loc (Pexp_ident (loc_lident lbl))
++let exp_of_label lbl =
++  Exp.mk ~loc:lbl.loc (Pexp_ident (loc_lident lbl))
+ 
+ let pat_of_label lbl =
+   Pat.mk ~loc:lbl.loc  (Ppat_var (loc_last lbl))
+ 
+ let mk_newtypes ~loc newtypes exp =
+@@ -822,13 +822,13 @@
+ %nonassoc HASH                         /* simple_expr/toplevel_directive */
+ %left     HASHOP
+ %nonassoc below_DOT
+ %nonassoc DOT DOTOP
+ /* Finally, the first tokens of simple_expr are above everything else. */
+-%nonassoc BACKQUOTE BANG BEGIN CHAR FALSE FLOAT INT
++%nonassoc BACKQUOTE BANG BEGIN CHAR FALSE FLOAT INT OBJECT
+           LBRACE LBRACELESS LBRACKET LBRACKETBAR LIDENT LPAREN
+-          NEW PREFIXOP STRING TRUE UIDENT
++          NEW PREFIXOP STRING TRUE UIDENT UNDERSCORE
+           LBRACKETPERCENT QUOTED_STRING_EXPR
+ 
+ 
+ /* Entry points */
+ 
+@@ -1309,10 +1309,13 @@
+         { (* TODO review mkmod location *)
+           Pmod_apply(me1, mkmod ~loc:$sloc (Pmod_structure [])) }
+     | (* An extension. *)
+       ex = extension
+         { Pmod_extension ex }
++    | (* A hole. *)
++      UNDERSCORE
++        { Pmod_hole }
+     )
+     { $1 }
+ ;
+ 
+ (* A parenthesized module expression is a module expression that begins
+@@ -2273,12 +2276,14 @@
+   | indexop_expr(qualified_dotop, expr_semi_list, LESSMINUS v=expr {Some v})
+     { mk_indexop_expr user_indexing_operators ~loc:$sloc $1 }
+   | expr attribute
+       { Exp.attr $1 $2 }
+ /* BEGIN AVOID */
++  (*
+   | UNDERSCORE
+      { not_expecting $loc($1) "wildcard \"_\"" }
++  *)
+ /* END AVOID */
+ ;
+ %inline expr_attrs:
+   | LET MODULE ext_attributes mkrhs(module_name) module_binding_body IN seq_expr
+       { Pexp_letmodule($4, $5, $7), $3 }
+@@ -2312,14 +2317,10 @@
+       { Pexp_for($3, $5, $7, $6, $9), $2 }
+   | ASSERT ext_attributes simple_expr %prec below_HASH
+       { Pexp_assert $3, $2 }
+   | LAZY ext_attributes simple_expr %prec below_HASH
+       { Pexp_lazy $3, $2 }
+-  | OBJECT ext_attributes class_structure END
+-      { Pexp_object $3, $2 }
```

Comment:
What is this diff about ?

Reply:
Same https://github.com/ocaml-ppx/ocamlformat/pull/1755